runtime/src/main/java/org/corfudb/protocols/service/CorfuProtocolMessage.java

```diff
@@ -131,6 +143,14 @@ public static RequestMsg getRequestMsg(HeaderMsg header, RequestPayloadMsg reque
      * @return           a ResponseMsg with the provided header and payload
      */
     public static ResponseMsg getResponseMsg(HeaderMsg header, ResponsePayloadMsg response) {
+        if (header.getVersion().getCorfuSourceCodeVersion() !=
+                GitRepositoryState.getCorfuSourceCodeVersion()) {
```

Comment:
Maybe this can be a private constant in the API class.

Reply:
Will do this in a following cleanup/refactor PR.

Reply:
Done, thanks!